insights/core/spec_factory.py

```diff
@@ -214,7 +214,22 @@ def __init__(self, relative_path, root="/", save_as=None, ds=None, ctx=None, cle
 
         self.validate()
 
+    def _is_inside_root(self):
```

Comment:
As this method is not used anywhere out the validate, better make it an `inner funciton` of the `validate`, Otherwise, we may need to add particular tests for this method (although it's a private one)

Reply:
P.S. The `FileProvider.validate()` method is not used only in `FileProvider.__init__`. Should it be defined as a nested function in the `__init__` method?

Reply:
Sorry, I didn't get your point

1. Although the `validate()` is called only in the `__ini__`, but for different child classes, it has different implementation. It should not be designed as inner function.
2. `inner function`, only works for function which is very simple (just like this one) and sometimes a bit common (can be called times) for its parent function.


If its function is common enough, and is useful not only for its parent caller. We can upgrade it as you did or might need to consider moving it to the `insights.util` as a utility function.

Because I didn't get from the code and feel hard to imagine the possibility that it will be used by other cases in this class, I asked you move it to a `inner function` or just expand it in the caller.